src/ffi.rs

```diff
@@ -0,0 +1,1775 @@
+//! Foreign Function Interface
```

Comment:
For your reference, PR #19 makes all the changes mentioned in the comments and it also adds a few extra functions for modifying citi::Record through the ffi.